developing-without-a-build/article.md

```diff
@@ -0,0 +1,266 @@
+```javascript
+import { kebabCase } from 'lodash-es';
+import kebabCase from 'lodash-es/kebabCase.js';
+```
+
+This doesn't work in the browser out of the box, as it doesn't know what file to request when it sees this path. Luckily there is a [new browser API called import maps](https://github.com/WICG/import-maps) which lets you instruct the browser how to resolve these imports:
+```html
+<script type="importmap">
+  {
+    "imports": {
+      "lodash-es": "./node_modules/lodash-es/lodash.js",
+      "lodash-es/": "./node_modules/lodash-es/"
+    }
+  }
+</script>
+```
+
+It's currently [implemented in chrome behind a flag](https://developers.google.com/web/updates/2019/03/kv-storage#import_maps), and it's easy to shim on other browsers [with es-module-shims](https://www.npmjs.com/package/es-module-shims). Until we get broad browser support, that can be an interesting option during development.
+
+If you don't want to go quite as cutting edge but still want to use the bare imports a great option is to use a web server which rewrites the bare imports on the fly into explicit paths before serving them to the browser. There are some servers available that do this. I recommend [es-dev-server](https://www.npmjs.com/package/es-dev-server), read more about it at the end of this article.
+
+### Caching
+Because we aren't bundling all of our code into just a few files, we don't have to set up any elaborate caching strategies. Your web server can use the file system's last modified timestamp to return a 304 if the file hasn't changed.
+
+You can test this in your browser by turning off `Disable cache` and refreshing:
+
+![](./images/caching.png)
+
+## Non-js modules
+So far we've only looked into javascript modules, and the story is looking pretty complete. It looks like we've got the tools needed to build different kinds of applications. But on the web we're not just writing javascript, we need to deal with other langauges as well.
+
+The good news is that there are proposals for HTML, CSS and JSON modules and all major browser vendors seem to be supportive of them:
+- [json modules](https://github.com/whatwg/html/issues/4315)
+- [html modules](https://github.com/w3c/webcomponents/issues/645)
+- [css modules](https://github.com/w3c/webcomponents/issues/759)
+
+The bad news is that they're not available just yet, and it will be a while for all browsers to support them. We have to look for some solutions in the meantime.
+
+### JSON
+In Node JS it's possible to import JSON files from javascript. These becomes available as javascript object. In web projects this is used frequently as well. There are many build tool plugins to make this possible.
+
+Until browsers support JSON modules, we can either just use a javascript module or use fetch to retreive the JSON files. See the `import.meta.url` section of an example of using fetch.
```

Comment:
```suggestion
Until browsers support JSON modules, we can either just use a javascript module or use fetch to retrieve the JSON files. See the `import.meta.url` section of an example of using fetch.
```

I Before E Except After C

Reply:
This got outdated, so I can't apply the changes anymore. Applied it manually, thanks!